tests/core/test_config.py

```diff
@@ -69,3 +72,58 @@ def check_form(p):
                 check_memoization(policy)
         elif isinstance(policy, FallbackPolicy):
             check_fallback(policy)
+
+
+TEST_MIGRATED_MAPPING_POLICIES = [
+    # no changes, no MappingPolicy
+    (
+        {"policies": [{"name": "MemoizationPolicy"}]},
+        {"intents": ["greet", "leave"]},
+        {
+            "config": {"policies": [{"name": "MemoizationPolicy"}]},
+            "domain_intents": ["greet", "leave"],
+            "rules": [],
+        },
+    ),
+    # MappingPolicy but no rules
+    (
+        {"policies": [{"name": "MemoizationPolicy"}, {"name": "MappingPolicy"}]},
+        {"intents": ["greet", "leave"]},
+        {
+            "config": {"policies": [{"name": "MemoizationPolicy"}]},
+            "domain_intents": ["greet", "leave"],
+            "rules": [],
+        },
+    ),
+    # MappingPolicy + rules
+    (
+        {"policies": [{"name": "MemoizationPolicy"}, {"name": "MappingPolicy"}]},
+        {
+            "intents": [{"greet": {"triggers": "action_greet"}}, "leave"],
+            "actions": ["action_greet"],
+        },
+        {
+            "config": {"policies": [{"name": "MemoizationPolicy"}]},
+            "domain_intents": ["greet", "leave"],
+            "rules": [
+                {
+                    "rule": "Rule to map `greet` intent (automatic conversion)",
+                    "steps": [{"intent": "greet"}, {"action": "action_greet"},],
+                }
+            ],
+        },
+    ),
+]
+
+
+@pytest.mark.parametrize("config,domain_dict,expected", TEST_MIGRATED_MAPPING_POLICIES)
+def test_migrate_mapping_policy_to_rules(
+    config: Dict[Text, Any], domain_dict: Dict[Text, Any], expected: Dict[Text, Any],
```

Comment:
optional: feels a bit clearer
```suggestion
    config: Dict[Text, Any], domain_dict: Dict[Text, Any], expected_config: Dict[Text, Any],
```

Reply:
I'll do `expected_results` because `expected_config["config"]` looks weird